mappings/net/minecraft/item/CrossbowItem.mapping

```diff
@@ -1,16 +1,83 @@
 CLASS bat net/minecraft/item/CrossbowItem
-	METHOD a getChargeSound (I)Lyj;
-		ARG 1 quickChargeLevel
+	FIELD c charged Z
+	FIELD d loaded Z
+	METHOD a getQuickChargeSound (I)Lyj;
+		ARG 1 stage
+	METHOD a getPullProgress (ILbcb;)F
+		ARG 0 useTicks
+		ARG 1 stack
+	METHOD a (Lahc;Lair;)V
+		ARG 1 e
+	METHOD a loadProjectiles (Lair;Lbcb;)Z
+		ARG 0 shooter
+		ARG 1 projectile
+	METHOD a loadProjectile (Lair;Lbcb;Lbcb;ZZ)Z
+		ARG 0 shooter
+		ARG 1 crossbow
+		ARG 2 projectile
+		ARG 3 simulated
+		ARG 4 creative
+	METHOD a (Lbbw;Lbcb;)Z
+		ARG 1 s
 	METHOD a hasProjectile (Lbcb;Lbbw;)Z
+		ARG 0 crossbow
+		ARG 1 projectile
 	METHOD a setCharged (Lbcb;Z)V
 		ARG 0 stack
 		ARG 1 charged
-	METHOD a shootAllProjectiles (Lbhi;Lair;Lahc;Lbcb;FF)V
+	METHOD a shootAll (Lbhi;Lair;Lahc;Lbcb;FF)V
+		ARG 0 world
+		ARG 1 entity
+		ARG 2 hand
+		ARG 3 stack
+		ARG 4 velocityMultiplier
+		ARG 5 trajectoryMultiplier
 	METHOD a shoot (Lbhi;Lair;Lahc;Lbcb;Lbcb;FZFFF)V
-	METHOD b storeChargedProjectile (Lbcb;Lbcb;)V
+		ARG 0 world
+		ARG 1 shooter
+		ARG 2 hand
+		ARG 3 crossbow
+		ARG 4 projectile
+		ARG 5 soundPitch
+		ARG 6 creative
+		ARG 7 velocityMultiplier
+		ARG 8 trajectoryMultiplier
+		ARG 9 simulated
+	METHOD a postShoot (Lbhi;Lair;Lbcb;)V
+		ARG 0 world
+		ARG 1 entity
+		ARG 2 stack
+	METHOD a createArrow (Lbhi;Lair;Lbcb;Lbcb;)Lawc;
+		ARG 0 world
+		ARG 1 entity
+		ARG 2 crossbow
+		ARG 3 arrow
+	METHOD a getSoundPitches (Ljava/util/Random;)[F
+		ARG 0 random
+	METHOD a getSoundPitch (Z)F
+		ARG 0 flag
+	METHOD b putProjectile (Lbcb;Lbcb;)V
+		ARG 0 crossbow
+		ARG 1 projectile
+	METHOD b (Lbcb;Lbhi;Lair;)F
+		ARG 0 stack
+		ARG 1 world
+		ARG 2 entity
+	METHOD c (Lbcb;Lbhi;Lair;)F
+		ARG 0 stack
+		ARG 1 world
+		ARG 2 entity
 	METHOD d isCharged (Lbcb;)Z
 		ARG 0 stack
+	METHOD d (Lbcb;Lbhi;Lair;)F
+		ARG 0 stack
+		ARG 1 world
+		ARG 2 entity
 	METHOD e getPullTime (Lbcb;)I
 		ARG 0 stack
-	METHOD j getChargedProjectiles (Lbcb;)Ljava/util/List;
+	METHOD j getProjectiles (Lbcb;)Ljava/util/List;
+		ARG 0 crossbow
 	METHOD k clearProjectiles (Lbcb;)V
+		ARG 0 crossbow
+	METHOD l getVelocityMultiplier (Lbcb;)F
```

Comment:
Would these `velocityMultiplier` names work as just `speed`?

Reply:
Possibly? It computes a multiplier that is applied to an existing projectile velocity. If you think speed is appropriate for that I will change it. See `ProjectileEntity#setVelocity`.

Reply:
```java
private static float getVelocityMultiplier(ItemStack stack) {
    if (stack.getItem() == Items.CROSSBOW && hasProjectile(stack, Items.FIREWORK_ROCKET)) {
        return 1.6f;
    }
    return 3.15f;
}
```
```java
new Vec3d(x, y, z).normalize().add(this.random.nextGaussian() * 0.007499999832361937 * trajectoryMultiplier, this.random.nextGaussian() * 0.007499999832361937 * trajectoryMultiplier, this.random.nextGaussian() * 0.007499999832361937 * trajectoryMultiplier).multiply((double)velocityMultiplier);
```

Reply:
I think `speed` works nicely here.
Also, for `trajectoryMultiplier`: could maybe be something like `inaccuracy` or `deviation`? Not sure, but the current name is fine too.

Reply:
`inaccuracy` was considered (I think it is also the name MCP opted for), but I don't believe it represents the parameter well. The default value for full precision is `1.0`, and if I saw an "inaccuracy" I would expect `0.0` to be full precision. It is very clearly a multiplier imo.

Reply:
Though, `1.0` wouldn't be full precision based on that usage?

Reply:
`1.0` wouldn't be full precision, though, would it? Maybe I'm missing something

Reply:
The following is appended to the velocity for a trajectory. A multiplier of `1.0` means no change:
`
.add(this.random.nextGaussian() * 0.008 * trajectoryMultiplier, this.random.nextGaussian() * 0.008 * trajectoryMultiplier, this.random.nextGaussian() * 0.008 * trajectoryMultiplier)
`

Reply:
With `1.0` would it not add velocity up to `0.008` on each axis?

Reply:
Oh.. Yeah, right... I'm bad at math. So, `inaccuracy` then?

Reply:
Yeah. Could also be like `inaccuracyMultiplier` or `inaccuracyFactor`.

Reply:
Resolved by b4a92fd after discussion in Discord